### PR TITLE
http/config: TransportSpec integration tests

### PR DESCRIPTION
**Pull request by abhinav**
This adds comprehensive tests for the HTTP TransportSpec.

### CODE REVIEW

**Comment by abhinav** [comment]
Changing to WIP now that #957 has landed.

File: transport/http/config_test.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+		opts []Option
+		yaml string
+
+		wantErrors    []string
+		wantInbound   *inbound
+		wantOutbounds outbounds
+	}{
+		{
+			desc:       "inbound without address",
+			yaml:       "inbounds: {http: {}}",
```
Comment (breerly) [inline transport/http/config_test.go]: whoah, json *in* yaml. *themoreyouknow*

File: transport/http/config_test.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+		{
+			desc: "inbound",
+			yaml: expand(`
+				inbounds:
+					http: {address: ":8080"}
+			`),
+			wantInbound: &inbound{Address: ":8080"},
+		},
+		{
+			desc: "inbound interpolation",
```
Comment (breerly) [inline transport/http/config_test.go]: This test case is so slick.

**Comment by abhinav** [comment]
This is no longer WIP

**Comment by breerly** [comment]
still lgtm

**Pull request merged**